### PR TITLE
Add scopes for Group models

### PR DESCRIPTION
**Pull request by andyksaw**
## Overview of Changes

Adds permission scopes for each group (many to many).
A scope determines what parts of the panel the group has access to.

### Gate

Group scopes utilise Laravel gates. Basically, each case inside the `GroupScope` enum is registered as a separate Laravel gate.

If you need to check for scope manually (eg. inside a controller) then you can do so via its gate:
```php
Gate::check(PanelGroupScope::MANAGE_ACCOUNTS->value)
```

### Middleware

You can use the in-built Gate middleware to limit a route to only users in a group that have the given scope:
```php
'middleware' => PanelGroupScope::ACCESS_PANEL->toMiddleware(),

// or

Route::midleware(PanelGroupScope::ACCESS_PANEL->toMiddleware())->group(...)
```

### Blade

Additionally adds a new blade view directive `@scope` which can be used to execute code only if the user's current group (if any) has a given scope, like so:
```php
@scope(PanelGroupScope::ACCESS_PANEL)
  // User has panel access
@elsescope
  // User does not have panel access
@endscope
```

### Caching

Scope fetching is a bit costly because of the one->many->many relationship. So the first scope lookup in a request will cache all of the scopes that belong to that user.

In other words, subsequent lookups in the same request will not require a fetch and are a simple dictionary key lookup.

## Deployment Requirements
- [ ] Composer update
- [ ] NPM update
- [x] Database migration
- [ ] .env update
- [ ] Environment upgrade (eg. PHP version)
- [ ] Other (please specify)

### CODE REVIEW

File: routes/web_panel.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+Route::group([
+    'prefix' => 'panel',
+    'as' => 'front.panel.',
+    'middleware' => [
+        'auth',
+        PanelGroupScope::ACCESS_PANEL->toMiddleware(),
+        'requires-mfa',
+    ],
+], function () {
```
Comment (freddyheppell) [inline routes/web_panel.php:33]: Great idea to split this out, have been meaning to do it for a while. You should be able to move all these `::group()` settings into the RouteServiceProvider so the routes in this file aren't nested

Comment (andyksaw) [reply routes/web_panel.php:33]: Good point, let me fix that 👍
